main/index.ts

```diff
@@ -34,26 +35,41 @@ process.env.BUNDLE_LOCATION = process.env.BUNDLE_LOCATION || path.resolve(__dirn
 log.transports.console.level = process.env.LOG_LEVEL || 'info'
 log.transports.file.level = ['development', 'test'].includes(process.env.NODE_ENV) ? false : 'verbose'
 
-function getCrashReportFields () {
+function getCrashReportFields() {
   const fields = ['networks', 'networksMeta', 'tokens']
 
   return fields.reduce((extra, field) => {
     return { ...extra, [field]: JSON.stringify(store('main', field) || {}) }
   }, {})
 }
 
+function getSentryException(event: Event) {
+  const frames = event?.exception?.values?.[0].stacktrace?.frames || []
+  const safeExceptionModule = ({ module = '' }) => {
+    const matches = /\((.+)[\\|\/]frame[\\|\/]resources[\\|\/]app.asar[\\|\/](.+)\)/.exec(module)
+    if (matches && matches[2]) {
+      return `{asar}/${matches[2].replaceAll('\\', '/')}`
+    }
+    return module
+  }
+  const safeFrames = frames.map((frame: StackFrame) => ({ ...frame, module: safeExceptionModule(frame) }))
+
+  return {
+    exception: { values: [{ stacktrace: { frames: safeFrames } }] },
+  }
+}
+
 Sentry.init({
   // only use IPC from renderer process, not HTTP
   ipcMode: Sentry.IPCMode.Classic,
   dsn: 'https://7b09a85b26924609bef5882387e2c4dc@o1204372.ingest.sentry.io/6331069',
-  beforeSend: (evt) => {
-    return {
+  beforeSend: (evt: Event, hint?: EventHint) => ({
       ...evt,
+      exception: getSentryException(evt),
       user: { ...evt.user, ip_address: undefined }, // remove IP address
       tags: { ...evt.tags, 'frame.instance_id': store('main.instanceId') },
-      extra: getCrashReportFields()
-    }
-  }
+      extra: getCrashReportFields(),
+    } as Event),
```

Comment:
don't need the type coercion here, can just do something like:
`exception: { values: getSentryExceptions(evt) }`

 and simply return the array of exceptions from that function